File: libgomp/testsuite/libgomp.c/task-7.c
/* { dg-do run } */

#include <omp.h>
#include <stdlib.h>

int
main ()
{
  #pragma omp task final (1)
  {
    if (!omp_in_final ())
      abort ();
    #pragma omp task
    {
      if (!omp_in_final ())
	abort ();
      #pragma omp target nowait
      if (omp_in_final ())
	abort ();
      if (!omp_in_final ())
	abort ();
      #pragma omp taskwait
    }
  }
  return 0;
}

File: libgomp/testsuite/libgomp.fortran/task-7.f90
! { dg-do run }

program main
  use omp_lib
  implicit none

  !$omp task final (.true.)
    if (.not. omp_in_final ()) &
      error stop
    !$omp task
      if (.not. omp_in_final ()) &
        error stop
      !$omp target nowait
      if (omp_in_final ()) &
        error stop
      !$omp end target
      if (.not. omp_in_final ()) &
        error stop
      !$omp taskwait
    !$omp end task
  !$omp end task
end
